docs/determinism.md

```diff
@@ -0,0 +1,13 @@
+# Determinism (GPU)
+
+## Compilation
+
+XLA is generally developed such that the compilation process is deterministic by choosing appropriate algorithms and data structures.
+
+One important exception is autotuning, in particular of GEMM fusions: due to fluctuations in measurements different kernels can be picked as the fastest ones in different compilation runs. [Persisted autotuning](./persisted_autotuning) can be used as a workaround.
+
+Another small exception is string serialization of protobufs with hash maps like fusion backend configs: these are not printed in deterministic order. This leads to variations in HLO IR dumps and can cause bugs if backend configs are used to compare HLO instructions. See the [helper methods](https://github.com/openxla/xla/blob/a02433bdc5e6dbe68446d6c95c4959b5e1715cf9/xla/service/gpu/ir_emission_utils.h#L242-L265) added to work around this problem.
```

Comment:
This conflates internal details and externally observable behavior. Are those actually compared ? Does the behavior depend on this comparison? Many things *can* be done, e.g. iterating over flat_hash_map

Reply:
Removed the section. Fixing the printing of backend configs is more difficult, I'll give you a link to a discussion.